client/lib/utils/coerce-values.js

```diff
@@ -1,3 +1,5 @@
+let coerceFormValues;
```

Comment:
I think either refactoring `coerceFormValues` to handle non-objects or refactoring the array items map callback to choose between `coerceFormValues` and `coerceValue` based on the `type` in `fieldSchema.items` would work.